### PR TITLE
Stop GLTF importer from forced skipping animation tracks automatically

### LINKED ISSUE #59150: Missing tracks in imported .glb (glTF) animations
### Godot version

4.0 alpha 4

### System information

Windows 11

### Issue description

I use animation from Mixamo. One animation clip per .glb file. Each file has same problem: left-hand fingers nodes have both, rotation and position tracks but right-hand fingers nodes have only position tracks, without rotation.
Importing animations to blender or different engines has no such issue.

Changing import options like "Optimizer", "Compression" or "Import Tracks" has no effect.
Re-exporting animations to .gltf format instead .glb has no effect too.

### Steps to reproduce

1. Import any of attached animation file
2. Drag&drop imported asset to a scene
3. Right click > Editable Children
4. Select AnimationPlayer
5. You'll see that e.g. LeftHandPinkyX have both position and rotation tracks, but equivalent RightHandPinkyX only position.
6. Now import file to Blender - all tracks exists as they should

### Minimal reproduction project

[anim.zip](https://github.com/godotengine/godot/files/8248218/anim.zip)

### PR DESCRIPTION
**Pull request by bartekd97**
Fixes #59150 

TL;DR:
GLTF improter as the only importer here was hardcoded to skip animation tracks that don't differ from rest pose. If you are going to reuse that animation on different (but compatible) target whose rest pose is different, you have broken animation.
No other importer do that and if we actually want to skip some tracks or optimize them, it's more reasonable to leave it to "Optimizer" and "Import Tracks" options, cause they are common for every improter.

### CODE REVIEW

**Comment by fire** [comment]
Reviewing - GDC is next week so I'll see what I can do.

**Comment by fire** [comment]
@bartekd97 can you attach a gltf file that has a "godot" skeleton to make my testing easier?

**Review by fire — APPROVED** [review]
`!transform_affects_skinned_mesh_instance` 

When you put animations that was skinned but is not anymore, it causes the code to use the node animation path.

When optimized in this way, it wasn't compatible with the skinned transforms.

Looks good to me.

@reduz I am not sure why you are optimizing at this stage. There's the compressed track and animation optimizer stack.

Edited:

These tracks are physically in the gltf2 file, so I would expect them to be present even if constant in the animation.

Edited 2:

This code causes all the tracks to be keyed that were animated.

Edited 3:

This code only affects node animations. The additional keyframes in compressed tracks will only be added on node animations.

image

image

Edited 4: 

The above photos are comparing 8 vs 32 page size.

**Comment by TokageItLab** [comment]
If this PR removes the ability to delete tracks, and the optimizer can do it instead, then there is no problem. However if not, then the method to delete the unnecessary tracks needs to be reimplemented somewhere.

**Comment by bartekd97** [comment]
@fire I'm uploading animation with godot skeleton and sample character mesh with skeleton for testing, if still needed.

[vampire_a_lusth.zip](https://github.com/godotengine/godot/files/8307413/vampire_a_lusth.zip)

**Comment by bartekd97** [comment]
> If this PR removes the ability to delete tracks, and the optimizer can do it instead, then there is no problem. However if not, then the method to delete the unnecessary tracks needs to be reimplemented somewhere.

I think that "Import Tracks" options should do the job. "Optimizer" as far as I know deletes mostly unnecessary keyframes.
image

I don't see a reason to forcely delete constant tracks automatically during import only for a one, specific file format - if they physically exist in file, then they exist for a reason and we should have them improted. Otherwise they wouldn't be exported at all.

**Comment by reduz** [comment]
@bartekd97 There are two reasons for this:
1) Saves space and processing time
2) If you want basic animation reuse, you generally want to omit the position tracks unless there is actual data in there. 

The importer uses the "IfPresentForall" to ensure the track will be there for all animations if any of the animations use it, so in practice even if the importer removes this, it is not a problem.

Regarding to "only the GLTF2 importer using it". The Collada importer will be deprecated soon and the FBX importer now works via FBX2GLTF2, hence it also uses the GLTF2 importer.

**Comment by akien-mga** [comment]
Based on the above comment, this change seems to be unwanted.
